plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -263,6 +273,10 @@ FrameUnwrap        : stack <| wrapped = case wrapped of
     VIWrap _ _ term -> stack <| term
     _               ->
         throwingWithCause _MachineError NonWrapUnwrappedMachineError $ Just $ ckValueToTerm wrapped
+FrameForce         : stack <| delayed = case delayed of
+    VDelay term -> stack |> term
+    _           ->
+        throwingWithCause _MachineError NonDelayedForcedError $ Just $ ckValueToTerm delayed
```

Comment:
Hm, in the untyped evaluator `force` can be applied to a builtin and here it can't. But we have stuff like `ifThenElse {() -> a} b x y ()` and it would be beneficial to use `delay`/`force` in pattern matching as well, given its ubiquity.

Reply:
Hm I guess that's true. I guess I should implement `forceEvaluate` also.

Reply:
Although now that I think about it: we don't need to apply force to builtins here, since it's the typed evaluator so we use type instantiations. So I'm not sure there is a need to apply force to anything other than a term that has been delayed.

Reply:
Oh, you're right, my example would become `force (ifThenElse {delayed a} b x y)` and here `force` is applied to the result of `ifThenElse` rather than the builtin itself. Yeah, my bad, I agree we don't need to do anything here.

Reply:
I'm seeing quite a few occurrences of `(force (builtin ifThenElse))` in  the erased scripts.  This is coming from type instantiation but unfortunately we can get rid of it because the untyped evaluator is still checking that forces occur in places where there's an instantiation in the type scheme.  We probably can't get rid of that, but it's a little annoying.  (Note that no computation can in fact occur in the case, unless perhaps the last item in the type scheme of a builtin is an instantiation, so you'd be fully applying a builtin to its arguments and then not doing anything until you instantiate the application.  Did we decide to prohibit that?)

Reply:
Yes, in the untyped machine we apply force to builtins to substitute for type applications. This is in the typed CK machine, though, so we don't need to worry about that case.